### ISSUE TITLE
KassiopeiaSimpleDefaults.REDUCE: attribute value can be a Set

### ISSUE THREAD

**Issue by sebfz1** [ChronixDB/chronix.kassiopeia#11]
Hi,

I guess `KassiopeiaSimpleDefaults.REDUCE` has an issue: on the first iteration we store a Set as the value, and add again a Set over next iterations, which leads to a Set of a Set of a Set, etc...

I believe correct code would looks like:

``` java
REDUCE = (collected, reduced) -> {
    collected.addAll(reduced.getTimestampsAsArray(), reduced.getValuesAsArray());

    // The collected attributes
    Map<String, Object> collectedAttributes = collected.getAttributesReference();

    // merge the attributes
    // we iterate over the copy
    for (HashMap.Entry<String, Object> entry : reduced.attributes().entrySet())
    {
        Set<Object> set = new HashSet<>();

        // Attribute to add
        String attribute = entry.getKey();
        Object value = entry.getValue();

        if (collectedAttributes.containsKey(attribute))
        {
            Object collectedValue = collectedAttributes.get(attribute);

            if (collectedValue instanceof Set<?>)
            {
                set.addAll((Set<?>) collectedValue);
            }
            else
            {
                set.add(collectedValue);
            }
        }

        set.add(value);
        collectedAttributes.put(attribute, set);
    }

    return collected;
};
```

Thanks,
Sebastien.

**Comment by FlorianLautenschlager** [comment]
Thx for reporting. I will figure it out and let you know.

**Comment by sebfz1** [comment]
Thanks Florian! Btw, what's the channel to ask general user questions?

**Comment by FlorianLautenschlager** [comment]
Fixed in 0.3
